app/Http/Controllers/v1/CategoryController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Http\Controllers\v1;
+
+use App\Http\Controllers\Controller;
+use App\Models\Category;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+
+class CategoryController extends Controller
+{
+    public function index()
+    {
+        $categories = Category::all();
+
+        return response()->json($categories, Response::HTTP_OK);
+    }
+
+    public function show($id)
+    {
+        $category = Category::find($id);
+
+        if (! $category) {
+            return response()->json(['error' => 'Category not found'], Response::HTTP_NOT_FOUND);
+        }
+
+        return response()->json($category, Response::HTTP_OK);
+    }
+
+    public function store(Request $request)
+    {
+        $validator = validator($request->all(), [
+            'name' => 'required|string|max:255',
+            // Add other validation rules for your fields
```

Comment:
// Add other validation rules for your fields ;)

Reply:
You can reference other Request types in the code base, so validation can be called on the requests with the
`$request->validated()` method. 
[reference](https://github.com/chrissantillan/UnlockEdv2/blob/17bbac8fb3ecad79bbbab65ab0468443e3148aa7/app/Http/Requests/StoreProviderPlatformRequest.php)

can also be generated with the `php artisan make:request CategoryRequest`